app/routes/event/Events.ts

```diff
@@ -0,0 +1,158 @@
+    const students: ICourseStudent[] = await getStudentsByCourseId(courseId);
+    const assignments: any = students.map((student: ICourseStudent) => {
+        const assignment = new AssignmentModel({
+            courseId,
+            deadlineDate: event.endDateTime,
+            mentorId: student.mentors,
+            score: 0,
+            status: AssignmentStatus.Assigned,
+            studentId: student.userId,
+            taskId: event.id,
+        });
+        return { insertOne: { document: assignment } };
+    });
+    await AssignmentModel.bulkWrite(assignments);
+};
+
+export const createDeleteEventsRoute = async (ctx: Router.IRouterContext) => {
+    const { id, eventType } = ctx.params;
+    try {
+        const query =
+            eventType === 'session' ? await SessionModel.findByIdAndRemove(id) : await TaskModel.findByIdAndRemove(id);
+        if (query !== null) {
+            await AssignmentModel.remove({ taskId: id });
+        } else {
+            setResponse(ctx, NOT_FOUND);
+            return;
+        }
+        setResponse(ctx, OK);
+    } catch (e) {
+        ctx.status = INTERNAL_SERVER_ERROR;
+        ctx.logger.error(e, 'Failed to remove document');
+    }
+};
+
+export const createPatchEventsRoute = async (ctx: Router.IRouterContext) => {
+    const { _id, ...body } = ctx.request.body;
+    const { eventType } = ctx.params;
+    try {
+        const result =
+            eventType === 'session'
+                ? await SessionModel.findByIdAndUpdate(_id, body, { new: true })
+                : await TaskModel.findByIdAndUpdate(_id, body, { new: true });
+        if (result === null) {
+            setResponse(ctx, NOT_FOUND);
+            return;
+        }
+
+        setResponse(ctx, OK, result);
+    } catch (e) {
+        ctx.status = INTERNAL_SERVER_ERROR;
+        ctx.logger.error(e, 'Failed to update document');
+    }
+};
+
+export const createGetEventsRoute = async (ctx: Router.IRouterContext) => {
+    const options: { useObjectId: boolean } = { useObjectId: true };
+    try {
+        if (connection.readyState !== STATES.connected) {
+            ctx.status = INTERNAL_SERVER_ERROR;
+            return;
+        }
+        const data =
+            (await SessionModel.findById(options.useObjectId ? Types.ObjectId(ctx.params.id) : ctx.params.id).exec()) ||
+            (await TaskModel.findById(options.useObjectId ? Types.ObjectId(ctx.params.id) : ctx.params.id).exec());
+        if (data === null) {
+            ctx.body = {};
+            ctx.status = NOT_FOUND;
+            return;
+        }
+        const body: IApiResponse<IEventModel> = {
+            data,
+        };
+        ctx.body = body;
+        ctx.status = OK;
+    } catch (err) {
+        ctx.logger.error(err);
+        ctx.status = INTERNAL_SERVER_ERROR;
+    }
+};
+
+const getStudentsByCourseId = async (courseId: string) => {
+    const result: ICourseStudent[] = await CourseStudentModel.aggregate([
+        {
```

Comment:
You needn't do aggregation here. To get userId and mentors you can use `CourseStudentModel.find({courseId})`

Reply:
Fixed.